### DISCUSSION TITLE
Create Item with Relational Data through API

### DISCUSSION THREAD

**Discussion by beschneider** [directus/directus#10450] [Archive]
Let's say I have a collection "houses" and a related collection "colors." Each house has a color.

When I create a new house entry via the API, am I only able to supply the color's primary key in the query? Or is there a way to create the house entry by giving Directus the **name** of the color, relying on Directus to find the key associated with that color to store in the "houses" collection?

Essentially I'm trying to skip a read query to find the color's primary key before making a query to store a new house with its color.

**Comment by benhaynes** [comment]
I've never heard of that feature on a platform, but it's interesting. Almost like a "fuzzy" create, not knowing our explicitly setting the FK.

An interesting concept. @rijkvanzanten have you heard of this before?

**Reply by beschneider** [reply]
Oh just to be clear, I haven’t heard of this elsewhere either. All of my past projects have interacted with databases with direct SQL queries, so all tasks were manual. I’m just starting to experiment with some platforms and database abstraction tools, so I’m just not sure to what level the gears turn under the hood for you. My fear is that I will still use platforms and tools very manually instead of fully utilizing their features, so that’s where my question comes from. Would be a cool feature though!

**Reply by rijkvanzanten** [reply]
Interesting! As of right now, when you pass a partial object _without_ the primary key, the nested item will be created as a totally new record, rather than trying to find a record with an existing value that matches.

**Reply by willywongi** [reply]
I stumble upon this discussion, to which I would like to add a request: what happens if I pass a nested object with the PK and attributes but the referenced item doesn't exist? Using the example from the original question (assuming the color collection has the rgb as the pk).
```json
{
  "floors": 2,
  "roof_type": "flat"
  "color": {
    "id": "FF0000",  # this nested item doesn't exist
    "name": "red"
  }
}
```
Will directus create the missing color or will raise a Not found error?

**Reply by rijkvanzanten** [reply]
> I stumble upon this discussion, to which I would like to add a request: what happens if I pass a nested object with the PK and attributes but the referenced item doesn't exist? Using the example from the original question (assuming the color collection has the rgb as the pk).
> 
> ```json
> {
>   "floors": 2,
>   "roof_type": "flat"
>   "color": {
>     "id": "FF0000",  # this nested item doesn't exist
>     "name": "red"
>   }
> }
> ```
> 
> Will directus create the missing color or will raise a Not found error?

It will create the item 👍🏻 The nested object is effectively an upsert

**Reply by willywongi** [reply]
Yay! You should add this piece of info in the docs!

**Reply by rijkvanzanten** [reply]
(cc @erondpowell ☝🏻)

**Reply by erondpowell** [reply]
on it bozz

**Comment by vonadz** [comment]
Does this work with the SDK?

**Reply by danielband** [reply]
Creating relational items? Yes, that works, [see](https://docs.directus.io/reference/introduction.html#relational-data).
What OP asked is probably not possible or at least I would not expect such behavior, because one won´t 'save' a query. Remember Directus is a wrapper for relational databasesand those don´t work that way. Also it can turn out to be an expensive operation, if the column is not an index.
